### PR TITLE
Chore: Remove global mock of `plugin_loader`

### PR DESCRIPTION
**Pull request by tomratcliffe**
**What is this feature?**
Removing this means we can get more accurate datasources behaviour in tests. With the mock removed, you can properly render components and test exported components etc.

**Why do we need this feature?**
In PRs such as https://github.com/grafana/grafana/pull/90363, we aren't able to test the dynamic loading of plugin/datasource components. Removing this mock lets us test more advanced patterns/bugs, and standardise our datasources approach

**Who is this feature for?**
UI devs

### CODE REVIEW

**Comment by jackw** [comment]
Nice that we're tackling some circular dependencies here! I also see no harm in removing the global mock either.

>In PRs such as https://github.com/grafana/grafana/pull/90363, we aren't able to test the dynamic loading of plugin/datasource components. Removing this mock lets us test more advanced patterns/bugs, and standardise our datasources approach

Have you considered using playwright to test such things as [dynamic plugin loading](https://github.com/grafana/grafana/blob/main/e2e/plugin-e2e/mssql/mssql.spec.ts)? The [plugin-e2e docs](https://grafana.com/developers/plugin-tools/e2e-test-a-plugin/test-a-data-source-plugin/) give some insights into how to approach testing various datasource features.

File: public/app/angular/panel/specs/metrics_panel_ctrl.test.ts [hunk 1]
```diff
@@ -2,18 +2,12 @@ jest.mock('app/core/core', () => ({}));
 jest.mock('app/core/config', () => {
   return {
     ...jest.requireActual('app/core/config'),
-    bootData: {
```
Comment (tomratcliffe) [inline public/app/angular/panel/specs/metrics_panel_ctrl.test.ts:5]: Here and in other places, we need to remove the partial-setting of config properties, as the general assumption is that `bootData.settings` is defined. 

With the `plugin_loader` mock removed, we now end up calling the real logic in `public/app/core/trustedTypePolicies.ts`, so we need to fixup some tests that ended up with an undefined `settings` property

**Review by konrad147 — APPROVED** [review]
LGTM! Let other teams to review the PR as well

**Review by ifrost — APPROVED** [review]
Explore changes LGTM 👍

**Pull request merged**